src/legacy/core_plugins/telemetry/index.ts

```diff
@@ -97,7 +101,13 @@ const telemetry = (kibana: any) => {
       mappings,
     },
     init(server: Server) {
-      const initializerContext = {} as PluginInitializerContext;
+      const initializerContext = {
+        env: {
+          packageInfo: {
+            version: getCurrentKibanaVersion(server),
+          },
+        },
+      } as PluginInitializerContext;
```

Comment:
I realise this was how it was before, so this isn't new in this PR, but I feel the use of `as` is dangerous.
It seems to tell TS not to typecheck `initializerContext` at all, meaning that when we then do `initializerContext.env.packageInfo.version` in `plugin.ts` we have no compile time guarantee that there isn't an `undefined` in there somewhere in the chain.

Is there a way for us to either provide all the props required by `PluginInitializerContext ` or be explicit about it being a `Partial<PluginInitializerContext>` (this will allow you to omit other props, but will ensure the props you _do_ provide are correctly typed.)

Reply:
`as` will type error if types are not compatible. Maybe another way is writing `: Partial< PluginInitializerContext >` but i havent tested that. to ignore compatiblity checks you'd have to do `as any as PluginInitializerContext`.

Reply:
ya, makes sense to fit `Partial<PluginInitializerContext>` in here, to type check the `env.packageInfo.version` bit, will require some later noise to do an `as PluginInitializerContext` somewhere else, or something :-)

Reply:
So, I tried using this:

```
      const initializerContext: Partial<PluginInitializerContext> = {
        env: {
          packageInfo: {
            version: getCurrentKibanaVersion(server),
          },
        },
      };
```

But `Partial<>` only goes one level deep,and up getting TS messages because `env` is missing some properties.  Is there a `PartialDeep<>` or such :-) (narrator: [yes, there is](https://stackoverflow.com/questions/41980195/recursive-partialt-in-typescript/41980288#41980288)).

And it turns out @Bamieh is right - we actually do get errors if the literal object doesn't match the `as`'d type - changing `env` to `envx` eg gives a TS error about `envx` not existing in the type.  So it seems like it's good enough as is.

But here's a recursive partial solution:

```
type RecursivePartial<T> = {
  [P in keyof T]?: RecursivePartial<T[P]>;
};
      ...
      const initializerContext: RecursivePartial<PluginInitializerContext> = {
        env: {
          packageInfo: {
            version: getCurrentKibanaVersion(server),
          },
        },
      };
      ...
      telemetryPlugin(initializerContext as PluginInitializerContext).setup(coreSetup);
```

Not sure it's worth the extra noise, but I think it perhaps _looks_ safer, if you don't know the `as` on the literal object ends up enforcing the types of the fields in it.

Reply:
I ended up leaving this as is, since we already got the type-checking we need.  If any other code will need to deal with this "partial" plugin init context, might be worth figuring out a different story, but seems fine now.